crates/fuel-core/src/graphql_api/api_service.rs

```diff
@@ -214,6 +226,66 @@ impl RunnableTask for Task {
     }
 }
 
+const REQUIRED_FUEL_BLOCK_HEIGHT_HEADER: &str = "REQUIRED_FUEL_BLOCK_HEIGHT";
+const CURRENT_FUEL_BLOCK_HEIGHT_HEADER: &str = "CURRENT_FUEL_BLOCK_HEIGHT";
+
+async fn required_fuel_block_height<B>(
+    req: AxumRequest<B>,
+    next: Next<B>,
+) -> impl IntoResponse {
+    let last_known_block_height: BlockHeight = LAST_KNOWN_BLOCK_HEIGHT
+    .get()
+    //Maybe too strict?
```

Comment:
I find it weird that this can be unset. We could always default to 0 as well, which would just mean that we're not in sync yet (which I presume we're not if this value hasn't been set).

Reply:
It will be unset until the first block is processed. But if we are going to fetch the value directly from the DB then it won't be the case anymore. I am waiting to hear what approach @xgreenx prefers before refactoring this bit